pkg/table/tables/test/partition/partition_test.go

```diff
@@ -299,7 +299,7 @@ func TestLocatePartition(t *testing.T) {
 		tk0.MustQuery("explain format = 'brief' select id, type from t where  type = 'WatchEvent';").Check(testkit.Rows(""+
 			`TableReader 2.00 root partition:watch_event data:Selection`,
 			`└─Selection 2.00 cop[tikv]  eq(test.t.type, "WatchEvent")`,
-			`  └─TableFullScan 3.00 cop[tikv] table:t keep order:false`))
+			`  └─TableFullScan 3.00 cop[tikv] table:t, partition:watch_event keep order:false`))
 	}
```

Comment:
before this bugfix. You can see the final phsical plan in the datasource.

<img width="583" alt="image" src="https://github.com/user-attachments/assets/bed13c2e-c62d-499b-adf4-2cb599dda0e3" />

after this bugfix. you will see 

<img width="603" alt="image" src="https://github.com/user-attachments/assets/3e002906-c381-4275-9f01-f6385ca84e6d" />

Reply:
and the partition define looks like this.

<img width="572" alt="image" src="https://github.com/user-attachments/assets/c458955b-8351-4d0a-a7b3-2665d7db696e" />

Reply:
so you scan the all partition before the bugfix. after this bugfix, we only scan the necessary partiton.

Reply:
I was wrong. It actually reads the specific partition, but this part is determined in the executor.